### PR TITLE
Fix wrong/inconsistent/empty categorical splits PUBDEV-3758

### PR DESCRIPTION
**Pull request by arnocandel**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/567)

### CODE REVIEW

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
will look at pyunit_checkpoint_new_category_in_response.py

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@h2o-ops please test!

**Comment by arnocandel** [comment]
@mmalohlava The only (consistently) failing test is the R CRAN test. And I'm not sure how my PR would affect that...

**Comment by arnocandel** [comment]
@h2o-ops please test!

File: h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public DrfMojoWriter() {}
   public DrfMojoWriter(DRFModel model) { super(model); }
 
   @Override public String mojoVersion() {
-    return "1.00";
+    return "0.10";
```
Comment (mmalohlava) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: Just for future changes: we should have mojo version defined at single place

Comment (tomkraljevic) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: You can't reduce the number.  There are mojos already in the field.

Comment (arnocandel) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: @st-pasha told me to do this. Maybe he can clarify his reasoning.

Comment (tomkraljevic) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: A good rule of thumb is:  for new (ie full merge from master, not a patch) stable releases, the mojo api number must stay the same or monotonically increase.

Comment (arnocandel) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: I'd be fine with "1.10"

Comment (arnocandel) [inline h2o-algos/src/main/java/hex/tree/drf/DrfMojoWriter.java]: @mmalohlava - I think the idea was that different algos could implement a different mojo API?

**Pull request merged**